Jay/docs/ansible (#5104)
gravitational/teleport: docs/5.0/admin-guide.md

## Ansible Integration

Ansible uses the OpenSSH client by default. This makes it compatible with
Teleport without any extra work, except configuring OpenSSH client to work with Teleport Proxy.

> There are two recommended scenarios:

Reviewer: I would recommend organizing this differently. 

There are two methods for OpenSSH clients to **connect** to Teleport Nodes; connecting to a Teleport Node directly over port 3022, or proxying through the Teleport Proxy over port 3023. 

There are two methods for OpenSSH clients to **authenticate** to Teleport Nodes; using Teleport credentials from a `tsh` session, or using a long lived SSH certificate exported from the Teleport Auth server.

1. Ansible connects directly to Teleport Node using built in ssh library over port 3022 and Teleport user credentials

2. Ansible connects to Teleport Node using the Teleport Proxy as a jumphost. This method uses Teleport user credentials over port 3022

> The configuration for openssh is as follows:

Reviewer 2: Use `OpenSSH` consistently?

Author: yes, will look over and make OpenSSH consistent across the document

> 1. Add hosts to know hosts in Ansible node

Reviewer 2: ```suggestion
1. Add hosts to known hosts in Ansible node
```

    * From Teleport Auth Node:
        ```bash
        $ tsh auth export --type=host
        ```

>     * Add the output on Ansible node `/.ssh/known_hosts` by modifying the node section to match the nodes you are reaching. (Note: wildcards are accepted)

Reviewer 2: ```suggestion
    * Add the output on Ansible node `/.ssh/known_hosts` by modifying the node section to match the nodes you are reaching (note: wildcards are accepted).
```

>     * Add the output on Ansible node `/.ssh/known_hosts` by modifying the node section to match the nodes you are reaching (note: wildcards are accepted).

Reviewer: ```suggestion
    * Add the output on Ansible node `/.ssh/known_hosts` by modifying the host name to match the nodes you are connecting to (note: wildcards are accepted).
```
The OpenSSH documentation calls this a host name, not a node. Suggest using the OpenSSH documentation nomenclature for alignment.

    ```bash

>     @cert-authority <NODE(S)> ssh-rsa AAA...REDACTED...TD9 logins=-teleport-nologin-f0d...REDACTED...fb1&type=host

Reviewer: ```suggestion
    @cert-authority <hostname.example.com|hostname|10.0.0.0> ssh-rsa AAA...REDACTED...TD9 logins=-teleport-nologin-f0d...REDACTED...fb1&type=host
```
Maybe this to clarify what's suggested in the previous sentence.

    ```
    * Modify ssh config file to include the proxy node and the desired Teleport nodes with `ProxyCommand` or `ProxyJump` parameters configured

> 1. Create identity file and extract to Ansible node

Reviewer: Given the above comment about restructuring the layout, I think this section will need to be updated to reflect the options. 

Examples:

Connection Type (include very simple and minimal ~/.ssh/config examples)
- Direct to Node
- Proxy to Node

Authentication Type
- `tsh` credentials
- `tctl auth sign` long lived SSH creds

    * From Teleport Auth Node:
        ```bash
        $ tctl auth sign --user=<teleport-user> --format=openssh --out=<file-name>
        ```
    * Extract both files created and add to `/.ssh/` on Ansible node

> 1. Enable `scp mode` in the Ansible config file (default is `/etc/ansible/ansible.cfg` ):

Reviewer: This is common between all connection and authentication types, so only mention it once, after you've given examples connection and authentication types.

        ``` bash
        scp_if_ssh = True
        ```

> 1. Configure your OpenSSH to connect to Teleport proxy and use `ssh-agent` socket:

Reviewer: ```suggestion
1. Configure your OpenSSH client to connect to Teleport proxy and use `ssh-agent` socket:
```

    ```bash
    $ eval `ssh-agent`
    $ tsh --proxy=root.example.com login
    # Verify the certs were added
    $ ssh-add -L
    ```

!!! tip "NOTE"

>     Additional Teleport and OpenSSH intergration information can be located [HERE](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)

Reviewer 2: ```suggestion
    Additional Teleport and OpenSSH integration information can be located [HERE](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)
```

>     Additional Teleport and OpenSSH integration information can be located [HERE](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)

Reviewer: ```suggestion
    Additional Teleport and OpenSSH integration information can be located [here](https://goteleport.com/teleport/docs/openssh-teleport/#using-openssh-client)
```
